### DISCUSSION TITLE
Charging device while using OTG. Tested devices and walkarounds.

### DISCUSSION THREAD

**Discussion by Pi4s3k** [feelfreelinux/octo4a#121] [General]
I tested couple devices to check if they are able to charge while connected to 3d printer. So far i managed to:
-Charge Samsung Galaxy Note 4 with Y-cable while using OTG
-Charge Samsung Galaxy Note 8 and Samsung Galaxy S7 with wireless charger while using OTG.

I encountered problem while trying to charge Samsung Galaxy Note 5, because phone keeps pausing charging when I connect the OTG cable.

Also after some research I found out that older Samsung Galaxy devices (up to S3 or S4 I belive), might be able to charge and use OTG by entering docking mode. It is achieved by installing 69,4k Ohm Resistor between 4th and 5th pin on micro usb port.

I read somewhere that ability to charge and use OTG can also be achieved by installing custom kernel. It would be great if someone could share some information about possible software walkarounds for the purpose of using Y-cable instead of wireless charger.

Please feel free to share your results with other devices.

**Comment by midnightsmith89** [comment]
I'll add the Pixel 2 XL and the samsung note 10+ both charge and print with a usb c hub attached.

**Comment by graphiste** [comment]
> Charge Samsung Galaxy Note 4 with Y-cable while using OTG

My galaxy Note 4 doesnt have power with Y cable. Are you using a special kernel or some other mod?

**Reply by midnightsmith89** [reply]
Nope, just a docking cable off amazon.  https://www.amazon.com/dp/B07L32B9C2/ref=cm_sw_r_tw_dp_RQ1904FD2XJ5C5F1TPR1?_encoding=UTF8&psc=1 

Some devices arent wired for OTG and Charging

**Comment by dietzm** [comment]
So far I found very few devices which charge while OTG is in use. Therefore I propose other solutions to the users of my App: See http://dietzm.de/blog/2022/01/14/gcodeprintr-3d-print-and-charge/

**Comment by KevinRS** [comment]
I'm thinking a table of devices is needed, with users submitting what works and what doesn't, including what cables or hubs were used. There's no way currently to get an idea if a specific device will work, unless you find a post by someone who had that exact device work, and no indication of what cables will and will not work with a device. 
For example I have a cheap onn tablet, that runs android 10 and has USB-C, wanted to try it but the Y-cable I got from amazon: while it allows connection to the printer if the charging cable is disconnected, doesn't seem to pass charging through, and doesn't allow connection to the printer when the charging cable is connected. The Y-cable seems to work fine with my phone.
So I have no way to know if the problem is the tablet, and it's just not going to work at all, or compatibility of the tablet with this specific cable. The battery dies too fast running octo4a without charging to even be worth figuring out if it would get successful prints.

**Comment by JMVS** [comment]
I'm dealing with this issue too. Running Octo4a on the OnePlus 3T stock Android phone I wasn't able to get both working (it either charges or does OTG). I don't know if a kernel modification can change that.
On a Sony Xperia XZ1 Compact with stock Android both OTG+Charging works right out of the box, altough I haven't tested Octo4a, I've only tested with a pendrive.

EDIT: Both phones are USB Type-C and this was done with a cheap USB Hub

**Comment by SamuranHassar** [comment]
Just to quickly share my findings on USB C devices.

Huawei P9 doesn't work.
Samsung Galaxy S20 FE works.
Huawei P30 Pro works.

The Hauwei P9 is listed as having USB Type-C 2.0, the P30 Pro as USB Type-3 3.1 and the S20 FE as USB Type-C 3.2, USB On-The-Go.

The OnePlus 3T mentioned above is listed as a USB Type-C 2.0.
The Sony Xperia XZ1 Compact is listed as USB Type-C 2.0; USB On-The-Go.
The Pixel 2XL is listed as USB Type-C 3.1.
The Galaxy Note 10+ is listed as USB Type-C 3.1.

(Information taken from gsmarena.com) 

Taking this information into account and using gsmarena to find USB information on different Type-C phones might be a good way to tell in advance if one will work. If it just says USB Type-C or USB Type-C 2.0 that might be a good hint it won't work. USB Type C 2.0; USB On-The-Go, USB Type-C 3.1 and USB Type-C 3.2 (I would also have to assume USB Type-C 3.0, USB On-The-Go on the LG G5) look like they will work.

**Comment by NicobobinusX** [comment]
Test results with a USB-C OTG Y-cable:
1.  FAILED - cannot both charge and connect to printer simultaneously:  
          Samsung Galaxy A01 w/Metro by Tmobile firmware.  
          OnePlus Nord N200
          Blu View 3

3. SUCCESS - can both charge and connect to printer simultaneously: 
          Samsung Galaxy S21 5g w/Tmobile firmware

**Comment by klausmaus** [comment]
Galaxy A5 with UGREEN USB-C to USB 3.0 PD Converter: SUCCESS
I have removed all other Apps (if possible) and i using only octoa and the internal cam.

**Reply by wjflier** [reply]
Just solder charge cable to+ and - pins in phone, no battery needed.  This frees up the usb port.

**Comment by midnightsmith89** [comment]
Would this not just bypass the circuit protection for charging? This could overload the phone circuit and essentially destroy it right?

**Reply by wjflier** [reply]
No, circuit protection reads from the battery, since there's no battery. it doesn't affect the phone,  Mines been on for several days.  Doesn't get hot, doesn't shut off.

**Comment by Hazron2** [comment]
Hi,

I've got two Samsung S7 Edges, Android Ver 8.0.0.  Octo installed fine and I down addins etc as per one of the YouTube vids. But while I could connect to my computer via a normal USB cable I couldn't get it to connect to the 3D printer, my computer, or pull up anything on the web when I typed the IP address from the Octo Android app.  I tried several OTG cables, a Y cable and a Targus powered USB hub, and the original Samsung USB adapter.  I checked various settings on the phone including the ones in the Developer Options. Nothing seems to get it connecting.

I see some S7's in the Excel list of phones on this site, and some worked and some didn't.  

Any ideas?

**Comment by sv0o** [comment]
with the new v 2.0.0 and some performance issues arising in my dual device setup, I thought I would try installing on one my old LG V10 phones.  Short story: it works!  Advantages of the V10 are more memory, faster cpu, 64GB of internal storage and a really, really good camera.  Of course, with the old microUSB port and the occasional battery dying because you still can't quick charge through a hub or OTG connector...I'm once again considering soldering a power supply directly to the battery terminals.  I haven't tested Octolapse yet.  That's the one thing that would put me over the edge.  I'm very intrigued about the pull request with the new camera stuff.  How can I test that?  I've suspected for a long time that these LG phones are using a lot of power on the video and snapshots and some camera stuff might even interfere with the phone's charging.

**Comment by lynxgenisys** [comment]
I hope Im not reviving a dead horse here, but 1st, Samsung S8, works. (I need to double check, but the hub I use has a couple usb ports, and HDMI out. I forget if the S8 will, but I know my old s21 and 23 and 24, you can charge, run a mouse and keyboard, and plug into a tv/moniter etc. (Ive use a playstation controller and hdmi out, while chargign and playing games. (point is, you can hook up a mouse and bigger screen and do everything on the phone itself easily.

2nd, and more importantly THERE IS AN APP.... called "Blackr" It adds a little circle to the screen, you can move where ever you want, and it blanks out the screen, cutting power use, heat, and screen burn in, AND keeps the phone awake. I had relegated my old S8 to a time-lapse camera, and with an app caled "Framelapse" I would litterally set it up and let it run for weeks at a time.

Anyways, Im brand new to the whole Octoprint/octolapse/klipper only just found out today I can run it off an old phone and dont need a rasberry!
(Im actually here to try and figure out why my flash on my phone intermitantly flashes now since installing otca-lapse. I think it has to do with the UI customizer, as it put the webcam window on every screen, and on viewing it activates the flash, so i think its just intermitant updating the view or something.... it did some othe wierd stuff too before that started tho... but its still takin the timelapse properly at least!)

**Reply by sv0o** [reply]
> I hope Im not reviving a dead horse here,....

no problem.  thanks for leads on blackr and framelapse.  do you have specific info on the hub you have working?
